python-backend/app/api/parties/party/resources/party_resource.py

```diff
@@ -72,77 +75,60 @@ def get(self, party_guid=None):
                 'total': pagination_details.total_results,
             }
 
-    def create_party_context(self, party_type_code, party_name, first_name):
-        party_context = {'party_type_code': party_type_code, 'party_name': party_name}
-        if party_type_code == PARTY_STATUS_CODE['per']:
-            if not first_name:
-                self.raise_error(400, 'Error: Party first name is not provided.')
-            party_exists = Party.find_by_name(first_name, party_name)
-            if party_exists:
-                self.raise_error(
-                    400, 'Error: Party with the name: {} {} already exists'.format(
-                        first_name, party_name))
-            party_context.update({
-                'first_name': first_name,
-            })
-        elif party_type_code == PARTY_STATUS_CODE['org']:
-            party_exists = Party.find_by_party_name(party_name)
-            if party_exists:
-                self.raise_error(
-                    400, 'Error: Party with the party name: {} already exists'.format(party_name))
-        else:
-            self.raise_error(400, 'Error: Party type is not provided.')
-        return party_context
-
     @api.expect(parser)
     @requires_role_mine_create
     def post(self, party_guid=None):
         if party_guid:
             self.raise_error(400, 'Error: Unexpected party id in Url.')
         data = PartyResource.parser.parse_args()
-        party_context = self.create_party_context(data['type'], data['party_name'],
-                                                  data['first_name'])
 
         try:
-            party = Party(
-                party_guid=uuid.uuid4(),
-                phone_no=data['phone_no'],
-                email=data['email'],
-                phone_ext=data.get('phone_ext'),
-                **self.get_create_update_dict(),
-                **party_context)
+            party = Party.create(data['party_name'],
+                                 data['email'],
+                                 data['phone_no'],
+                                 data['type'],
```

Comment:
Will this send back a stack trace if one of these is missing? I believe we want a controlled error message.

Reply:
This should reflect all existing behaviour. I didn't feel comfortable refactoring these in this PR. This shows up in the diff, but the key access style of these properties has not been changed.

I would argue that we should convert these to use `.get()` for key access and check with an appropriate error, but I don't think it's appropriate in this PR. There's already scope creep happening.

All of the new functionality that I add has used `.get()` and proper null checks.

 Hope that seems reasonable

Reply:
I'm going to move the `AssertionError` `try` block wrap the `save()`, as I believe your other comment is correct. And I'll catch the key error here